### PR TITLE
Json-pushes version 2

### PR DESCRIPTION
**Pull request by camd**
This fixes [Bug 1076826](https://bugzilla.mozilla.org/show_bug.cgi?id=1076826)

This moves us to using ``version=2`` with our calls to json-pushes to fetch ``resultsets`` from Mercurial.  In doing this, we can call to the server more often than once perminute.  GPS says every 15 seconds is fine.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/881)

### CODE REVIEW

File: treeherder/settings/base.py [hunk 1]
```diff
@@ -206,9 +206,9 @@
 CELERY_DEFAULT_ROUTING_KEY = 'default'
 
 CELERYBEAT_SCHEDULE = {
-    'fetch-push-logs-every-minute': {
+    'fetch-push-logs-every-10-seconds': {
         'task': 'fetch-push-logs',
-        'schedule': timedelta(minutes=1),
+        'schedule': timedelta(seconds=10),
```
Comment (edmorley) [inline treeherder/settings/base.py]: This is fixing bug 1096919, rather than bug 1076826 - could we save that for bug 1096919?
I think it might be safer to change less at one time, and reducing the interval has additional dependencies (like the task locking):
https://bugzilla.mozilla.org/showdependencytree.cgi?id=1096919&maxdepth=1&hide_resolved=1

Comment (camd) [inline treeherder/settings/base.py]: sure, good call.  :)

**Comment by edmorley** [comment]
I forget now what was broken on stage. Has that since been fixed? If so, could you add a test for it?

**Comment by edmorley** [comment]
There's a lot of pre-existing repetition of `/json-pushes/?full=1` etc, would you mind filing a bug for refactoring some of that at a later date?

---

Reviewed 2 of 6 files at r1, 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/etl/pushlog.py, line 87 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxVzVFoCY-ed-lD8koo)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L87)):</sup>
s/revision/start ID/ (plus maybe a bit of a rewording of that sentence?). 

---

<sup>**[treeherder/etl/pushlog.py, line 95 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxVzj5XlrMCc2SDV3Cg)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L95)):</sup>
s/later/lower/
Also I think this wording might be clearer?
`"Got a ``lastpushid`` value of {}, which is lower than "`

Not sure if it's worth mentioning why this might have occurred? (eg due to repo reset)

---

<sup>**[treeherder/etl/pushlog.py, line 97 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxW1gW95XyPnM0E9cKi)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L97)):</sup>
"Getting all changes" implies we'll fetch all pushes for the repo, when in fact we'll only fetch the most recent N changesets.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/881)

**Comment by camd** [comment]
Good point.  OK, I entered: https://bugzilla.mozilla.org/show_bug.cgi?id=1198026

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/881)

**Comment by camd** [comment]
Review status: 7 of 8 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[treeherder/etl/pushlog.py, line 87 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxVzVFoCY-ed-lD8koo)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L87)):</sup>
Done.

---

<sup>**[treeherder/etl/pushlog.py, line 95 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxVzj5XlrMCc2SDV3Cg)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L95)):</sup>
Done.

---

<sup>**[treeherder/etl/pushlog.py, line 97 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxW1gW95XyPnM0E9cKi)** ([raw file](https://github.com/mozilla/treeherder/blob/9c9c8b8b10e75a1fe2e6242567d16d45a881f2fa/treeherder/etl/pushlog.py#L97)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/881)

**Comment by edmorley** [comment]
Just noticed the New Relic exceptions on the stage test. I believe it's from when pushes is empty (ie no new changes. How about:
```python
pushes = extracted_content['pushes']

if not pushes:
    return None

last_push_id = max(map(lambda x: int(x), pushes.keys()))
...
...

return top_revision
```

**Comment by edmorley** [comment]
Could you add a test for the empty pushes case? 
```json
{
"lastpushid": 12345,
"pushes": {}
}
```

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[treeherder/etl/pushlog.py, line 102 \[r4\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxX781hDwUcnXSOXihq)** ([raw file](https://github.com/mozilla/treeherder/blob/1551c131acede6d5046eb1c7e5bedc8415d685e9/treeherder/etl/pushlog.py#L102)):</sup>
s/later/lower/ or similar, is saying the opposite of what it means at the moment :-)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/881)

**Comment by camd** [comment]
Yep, done

---

Review status: 6 of 8 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

<sup>**[treeherder/etl/pushlog.py, line 102 \[r4\]](https://reviewable.io:443/reviews/mozilla/treeherder/881#-JxX781hDwUcnXSOXihq)** ([raw file](https://github.com/mozilla/treeherder/blob/1551c131acede6d5046eb1c7e5bedc8415d685e9/treeherder/etl/pushlog.py#L102)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/881)
